### PR TITLE
Tdfoptimisation

### PR DESCRIPTION
**Pull request by SteffenHeu**
Did some magic.

Before (import + mass detection): (about 700mb total ram usage, about 580 due to the data file)
grafik

Now: (about 300 mb total ram usage)
grafik

Some processing (before):
grafik

Some processing (now):
grafik

Also made a super low ram usage implementation that loads mobility scan data on demand, but i think it's too hard to optimise to keep. Will probalby remove it after some more testing tomorrow.

### CODE REVIEW

**Comment by Annexhc** [comment]
Looks very nice. Do you see any performance issues? Is the processing now slower?

**Comment by SteffenHeu** [comment]
No, it's still blazingly fast. The data is still loaded and memory mapped as usual, just the MobilityScan and MassList are not created anymore. They are now generated on demand instead. 
This implies that MergedMsMs will have a list of unique MobilityScans, which will be generated every time a new MsMsSpectrum is created. But still uses heaps less ram than previously.

On the other hand, using the super low ram option that loads (uncompressesing and converting from raw data on demand) has huge performance implications, which is why i will probably not keep it.

**Comment by robinschmid** [comment]
> No, it's still blazingly fast.

Blazingly fast is always a great description

File: src/main/java/io/github/mzmine/datamodel/impl/SimpleIsotopePattern.java [hunk 1]
```diff
@@ -234,6 +234,18 @@ public Double getBasePeakIntensity() {
     }
   }
 
+  @Override
+  public void getMzValues(double[] dst, int offset) {
```
Comment (robinschmid) [inline src/main/java/io/github/mzmine/datamodel/impl/SimpleIsotopePattern.java:238]: Do we actually need these two methods everywhere? Who is actually calling it? Maybe its enough to have it in the ion mobility specific classes? 

But you understand better the impact of that

File: src/main/java/io/github/mzmine/modules/dataprocessing/featdet_massdetection/MassDetectionTask.java [hunk 2]
```diff
@@ -154,18 +153,12 @@ public void run() {
         // [mzs, intensities]
         double[][] mzPeaks = detector.getMassValues(data, massDetector.getParameterSet());
 
-        if (scan instanceof Frame) {
+        if (scan instanceof SimpleFrame frame) {
```
Comment (robinschmid) [inline src/main/java/io/github/mzmine/modules/dataprocessing/featdet_massdetection/MassDetectionTask.java:156]: Why a SimpleFrame and not Frame?
Maybe getMobilityScanStorage should be pulled to Frame then?
or check in a different way?

File: src/main/java/io/github/mzmine/modules/io/import_rawdata_bruker_tdf/TDFImportModule.java [hunk 3]
```diff
@@ -44,10 +44,9 @@
  */
 public class TDFImportModule implements MZmineProcessingModule {
 
-  private Logger logger = Logger.getLogger(this.getClass().getName());
-
   private static final String MODULE_NAME = "Bruker TDF file import";
   private static final String MODULE_DESCRIPTION = "This module imports raw data into the project.";
+  private Logger logger = Logger.getLogger(this.getClass().getName());
```
Comment (robinschmid) [inline src/main/java/io/github/mzmine/modules/io/import_rawdata_bruker_tdf/TDFImportModule.java:49]: static

File: src/main/java/io/github/mzmine/modules/dataprocessing/featdet_massdetection/MassDetectionTask.java (hunk 2 above)
Comment (SteffenHeu) [reply src/main/java/io/github/mzmine/modules/dataprocessing/featdet_massdetection/MassDetectionTask.java:156]: We dont really need to expose the storage anyhwere else, so i don't see why it should be in the frame interface.

Comment (SteffenHeu) [reply src/main/java/io/github/mzmine/modules/dataprocessing/featdet_massdetection/MassDetectionTask.java:156]: Maybe i can solve two issues at once, give me a second.

File: src/main/java/io/github/mzmine/datamodel/impl/SimpleFrame.java [hunk 4]
```diff
@@ -103,10 +99,17 @@ public MobilityType getMobilityType() {
     return Range.singleton(0.0);
   }
 
+  public MobilityScanStorage getMobilityScanStorage() {
+    if(mobilityScanStorage == null) {
+      throw new IllegalStateException("Mobility scans not loaded");
```
Comment (robinschmid) [inline src/main/java/io/github/mzmine/datamodel/impl/SimpleFrame.java:104]: Maybe add some javadoc and improve the error message?
When is it usally loaded?
Why is it null?
What do we ahve to do first?

File: src/main/java/io/github/mzmine/modules/io/import_rawdata_bruker_tdf/TDFImportTask.java [hunk 5]
```diff
@@ -72,14 +75,13 @@
...
-  private final ParameterSet ms1DetectorParam;
-  @Nullable
-  private final ParameterSet ms2DetectorParam;
+  @Nullable private final MassDetector ms1Detector;
+  @Nullable private final MassDetector ms2Detector;
+  @Nullable private final ParameterSet ms1DetectorParam;
+  @Nullable private final ParameterSet ms2DetectorParam;
+
+  private final boolean denoising = false;
+  private static final double NOISE_THRESHOLD = 9E0;
```
[the remark this answers is not in the document]
Comment (SteffenHeu) [reply src/main/java/io/github/mzmine/modules/io/import_rawdata_bruker_tdf/TDFImportTask.java:84]: It's deactivated und we have extended preferences.

File: src/main/java/io/github/mzmine/datamodel/impl/SimpleFrame.java (hunk 4 above)
Comment (SteffenHeu) [reply src/main/java/io/github/mzmine/datamodel/impl/SimpleFrame.java:104]: during file import, so this shouldnt occur.  It's more of an implication for us that somethign went wrong.

**Comment by robinschmid** [comment]
All ready

**Pull request merged**